### DISCUSSION TITLE
IIS Hosting on Windows Server 2016?

### DISCUSSION THREAD

**Discussion by RickTheHat** [pocketbase/pocketbase#1096] [Q&A]
I'm extremely impressed with pocketbase and I have my own lightsail VM on windows where I have a few websites.

I'd like to host this on IIS and create a couple API backends with pocketbase.

Before I go down the rabbit hole of testing it out on IIS/Windows, wanted to know if anyone has already gotten there before me.

If so, tips? tricks? How was your experience? OR is it not possible to host on IIS?

**Comment by pchasco** [comment]
What is the specific question? I see no reason why it wouldn’t run on Windows Server 2016. It is a standalone server, so you wouldn’t host it within IIS. You’ll need to expose it on an available TCP port and open your firewall to enable outside connections, or if you have a reverse proxy, proxy inbound to it from that endpoint.

**Reply by pchasco** [reply]
My suggestion would be to host it on Linux and configure it as a service so that it will be restarted if it crashes for whatever reason. You can rent a capable Linux VM for a few bucks a month from many places. Or you could host it in Azure on their free tier App Service Plan. You would need to configure the path to store the volatile data (database) in persistent storage.

**Comment by RickTheHat** [comment]
@pchasco That's the kind of answer I was looking for as I wasn't sure if I would I host it under IIS or what you said does make sense to host it outside of IIS. I'm not versed in Linux so if I did get it installed on a Linux VM, it would be a win but after that, I'd be lost. I do see that [Lightsail](https://aws.amazon.com/lightsail/pricing/) has some cheap hosting so I could give it the old college try.

**Comment by JohnTing** [comment]
(Translated by ChatGPT.)

I am trying to use Packetbase on IIS in Windows 11, and have tried to start Packetbase server using both AspNetCoreModuleV2 and HttpPlatformHandler. However, I am getting the same error.

```
panic: no console

goroutine 1 [running]:
modernc.org/libc.newFile(0x15ea780?, 0x244e1f8?)
	/home/runner/go/pkg/mod/modernc.org/libc@v1.22.5/libc_windows.go:293 +0xb7
modernc.org/libc.init()
	/home/runner/go/pkg/mod/modernc.org/libc@v1.22.5/libc.go:56 +0x3831
```

These are my web.config

AspNetCoreModuleV2 
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <location path="." inheritInChildApplications="false">
    <system.webServer>
      <handlers>
        <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModuleV2" resourceType="Unspecified" />
      </handlers>
      <aspNetCore processPath=".\pocketbase.exe" arguments="serve" stdoutLogEnabled="true" stdoutLogFile=".\stdout" />
    </system.webServer>
  </location>
</configuration>
```


httpPlatformHandler
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.webServer>
    <handlers>
      <add name="httpplatformhandler" path="*" verb="*" modules="httpPlatformHandler" resourceType="Unspecified"/>
    </handlers>
    <httpPlatform processPath=".\pocketbase.exe" arguments="serve" stdoutLogEnabled="true" stdoutLogFile="./log.txt">
      <environmentVariables>
      </environmentVariables>
    </httpPlatform>
  </system.webServer>
</configuration>
```

**Reply by ganigeorgiev** [reply]
@JohnTing Unfortunately I'm not familiar with IIS and I'm not sure that I can help, but the [above go `panic`](https://gitlab.com/cznic/libc/-/blob/master/libc_windows.go#L291-L295) indicates that probably you don't have access to the stdin.

**Reply by JohnTing** [reply]
@ganigeorgiev  Thank you for your response. I am not familiar with IIS either. I think I will switch to nginx instead.

**Comment by pchasco** [comment]
I would ask why you want to manage PocketBase under IIS in the first place?
PocketBase is its own web server and does not need to be hosted by IIS. I
think you are just asking for problems in attempting to do that. If you
really wanted to expose PocketBase through IIS for some reason, you could
configure IIS as reverse proxy and forward requests to a locally running
instance of PocketBase.

**Reply by JohnTing** [reply]
(Translated by Google)
Initially, I hoped to manage the startup/shutdown of PocketBase instances on IIS and provide HTTPS. 
Due to some peculiar errors, I was unable to install ARR for reverse proxy on IIS. 
In the end, I switched to using nginx to provide HTTPS and reverse proxy.

**Reply by pchasco** [reply]
That’s a much easier approach. FYI PocketBase does support automatically provisioning SSL certificates with Let’s Encrypt, so you would not need a proxy to secure it with SSL. However, there are other benefits to using Nginx beyond terminating SSL. It’s probably how I would host it in production.
